x/bank/keeper/keeper.go

```diff
@@ -53,14 +55,26 @@ type BaseKeeper struct {
 	paramSpace paramtypes.Subspace
 }
 
-func (k BaseKeeper) GetTotalSupply(ctx sdk.Context) sdk.Coins {
-	balances := sdk.NewCoins()
-	k.IterateTotalSupply(ctx, func(balance sdk.Coin) bool {
-		balances = balances.Add(balance)
-		return false
+func (k BaseKeeper) GetPaginatedTotalSupply(ctx sdk.Context, pagination *query.PageRequest) (sdk.Coins, *query.PageResponse, error) {
+	store := ctx.KVStore(k.storeKey)
+	supplyStore := prefix.NewStore(store, types.SupplyKey)
+
+	supply := sdk.NewCoins()
+
+	pageRes, err := query.Paginate(supplyStore, pagination, func(key, value []byte) error {
+		amount, ok := sdk.NewIntFromString(string(value))
```

Comment:
?? why the amount is stored as a string??

Reply:
fwiu it's because `sdk.Int` doesn't implement proto message. @jgimeno should know better

Reply:
but it has `Bytes()` tho

Reply:
true, opened a pr for this https://github.com/cosmos/cosmos-sdk/pull/9051